### ISSUE TITLE
Unable to perform initial `chef-server-ctl reconfigure` on CentOS 6 & 7

### ISSUE THREAD

**Issue by rhass** [chef/chef-server#1574]
## Current Behavior
There are two failures which occur during the initial reconfigure:

```
Recipe: private-chef::bootstrap
  * execute[/opt/opscode/bin/chef-server-ctl start postgresql] action run
    - execute /opt/opscode/bin/chef-server-ctl start postgresql
  * execute[/opt/opscode/bin/chef-server-ctl start oc_bifrost] action run
    - execute /opt/opscode/bin/chef-server-ctl start oc_bifrost
  * ruby_block[bootstrap-chef-server-data] action run
    
    ================================================================================
    Error executing action `run` on resource 'ruby_block[bootstrap-chef-server-data]'
    ================================================================================
    
    Errno::ECONNREFUSED
    -------------------
    Failed to open TCP connection to 127.0.0.1:9463 (Connection refused - connect(2) for "127.0.0.1" port 9463)
    
    Cookbook Trace:
    ---------------
    /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:199:in `bifrost_request'
    /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:166:in `create_object_in_authz'
    /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:158:in `create_actor_in_authz'
    /var/opt/opscode/local-mode-cache/cookbooks/private-chef/libraries/chef_server_data_bootstrap.rb:40:in `bootstrap'
    /var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:36:in `block (2 levels) in from_file'
    
    Resource Declaration:
    ---------------------
    # In /var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb
    
     34: ruby_block "bootstrap-chef-server-data" do
     35:   block do
     36:     ChefServerDataBootstrap.new(node).bootstrap
     37:   end
     38:   not_if { OmnibusHelper.has_been_bootstrapped? }
     39:   notifies :restart, 'service[opscode-erchef]'
     40: end
     41: 
    
    Compiled Resource:
    ------------------
    # Declared in /var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:34:in `from_file'
    
    ruby_block("bootstrap-chef-server-data") do
      action [:run]
      default_guard_interpreter :default
      declared_type :ruby_block
      cookbook_name "private-chef"
      recipe_name "bootstrap"
      block #<Proc:0x00000000042b8820@/var/opt/opscode/local-mode-cache/cookbooks/private-chef/recipes/bootstrap.rb:35>
      block_name "bootstrap-chef-server-data"
      not_if { #code block }
    end
    
    System Info:
    ------------
    chef_version=14.3.37
    platform=centos
    platform_version=6.10
    ruby=ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
    program_name=/opt/opscode/embedded/bin/chef-client
    executable=/opt/opscode/embedded/bin/chef-client
    
Recipe: private-chef::postgresql
  * runit_service[postgresql] action restart (up to date)
  * execute[restart_postgresql_log_service] action run
    - execute /opt/opscode/embedded/bin/sv restart /opt/opscode/sv/postgresql/log
Recipe: private-chef::oc_bifrost
  * runit_service[oc_bifrost] action restart (up to date)
  * execute[restart_oc_bifrost_log_service] action run
    - execute /opt/opscode/embedded/bin/sv restart /opt/opscode/sv/oc_bifrost/log
Recipe: private-chef::oc_id
  * runit_service[oc_id] action restart (up to date)
  * execute[restart_oc_id_log_service] action run
    - execute /opt/opscode/embedded/bin/sv restart /opt/opscode/sv/oc_id/log
Recipe: private-chef::nginx
  * runit_service[nginx] action restart
    
    ================================================================================
    Error executing action `restart` on resource 'runit_service[nginx]'
    ================================================================================
    
    Mixlib::ShellOut::ShellCommandFailed
    ------------------------------------
    Expected process to exit with [0], but received '1'
    ---- Begin output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx ----
    STDOUT: fail: /opt/opscode/service/nginx: unable to change to service directory: file does not exist
    STDERR: 
    ---- End output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx ----
    Ran /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx returned 1
    
    Cookbook Trace:
    ---------------
    /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:174:in `tap'
    /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:174:in `safe_sv_shellout!'
    /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/helpers.rb:202:in `restart_service'
    /var/opt/opscode/local-mode-cache/cookbooks/runit/libraries/provider_runit_service.rb:330:in `block in <class:RunitService>'
    
    Resource Declaration:
    ---------------------
    # In /var/opt/opscode/local-mode-cache/cookbooks/enterprise/definitions/component_runit_service.rb
    
     37:   runit_service component do
     38:     action :enable
     39:     retries 20
     40:     control params[:control] if params[:control]
     41:     options(
     42:       log_directory: log_directory
     43:     )
     44:     params[:runit_attributes].each do |attr_name, attr_value|
     45:       send(attr_name.to_sym, attr_value)
     46:     end
     47:   end
     48: 
    
    Compiled Resource:
    ------------------
    # Declared in /var/opt/opscode/local-mode-cache/cookbooks/enterprise/definitions/component_runit_service.rb:37:in `block in from_file'
    
    runit_service("nginx") do
      params {:package=>"private_chef", :log_directory=>nil, :svlogd_size=>nil, :svlogd_num=>nil, :ha=>nil, :control=>nil, :runit_attributes=>{}, :action=>:enable, :name=>"nginx"}
      provider Chef::Provider::RunitService
      action [:enable]
      default_guard_interpreter :default
      service_name "nginx"
      enabled false
      running false
      masked nil
      options {:log_directory=>"/var/log/opscode/nginx"}
      pattern "nginx"
      status_command "/opt/opscode/embedded/bin/sv status /opt/opscode/service"
      supports {:restart=>true, :reload=>true, :status=>true}
      sv_bin "/opt/opscode/embedded/bin/sv"
      sv_dir "/opt/opscode/sv"
      service_dir "/opt/opscode/service"
      lsb_init_dir "/opt/opscode/init"
      log true
      restart_on_update true
      run_template_name "nginx"
      log_template_name "nginx"
      check_script_template_name "nginx"
      finish_script_template_name "nginx"
      sv_templates true
      log_dir "/var/log/nginx"
      service_mirror # Declared in 
    
    service("nginx") do
      provider Chef::Provider::Service::Simple
      action [:nothing]
      default_guard_interpreter :default
      service_name "nginx"
      enabled nil
      running nil
      masked nil
      pattern "nginx"
      start_command "/opt/opscode/embedded/bin/sv start /opt/opscode/service/nginx"
      stop_command "/opt/opscode/embedded/bin/sv stop /opt/opscode/service/nginx"
      status_command "/opt/opscode/embedded/bin/sv status /opt/opscode/service/nginx"
      restart_command "/opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx"
      supports {:restart=>true, :reload=>true, :status=>true}
    end
    
      declared_type :runit_service
      cookbook_name "private-chef"
      recipe_name "nginx"
      retries 20
    end
    
    System Info:
    ------------
    chef_version=14.3.37
    platform=centos
    platform_version=6.10
    ruby=ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux]
    program_name=/opt/opscode/embedded/bin/chef-client
    executable=/opt/opscode/embedded/bin/chef-client
    
Recipe: private-chef::opscode-solr4
  * runit_service[opscode-solr4] action restart (up to date)
  * execute[restart_opscode-solr4_log_service] action run
    - execute /opt/opscode/embedded/bin/sv restart /opt/opscode/sv/opscode-solr4/log
Recipe: private-chef::opscode-expander
  * runit_service[opscode-expander] action restart (up to date)
  * execute[restart_opscode-expander_log_service] action run
    - execute /opt/opscode/embedded/bin/sv restart /opt/opscode/sv/opscode-expander/log
Recipe: private-chef::bookshelf
  * runit_service[bookshelf] action restart (up to date)
  * execute[restart_bookshelf_log_service] action run
    - execute /opt/opscode/embedded/bin/sv restart /opt/opscode/sv/bookshelf/log
Recipe: private-chef::opscode-erchef
  * runit_service[opscode-erchef] action restart (up to date)
  * execute[restart_opscode-erchef_log_service] action run
    - execute /opt/opscode/embedded/bin/sv restart /opt/opscode/sv/opscode-erchef/log

Running handlers:
Running handlers complete
Chef Client failed. 264 resources updated in 02 minutes 27 seconds
[2018-09-12T12:51:13-04:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
[2018-09-12T12:51:13-04:00] FATAL: Stacktrace dumped to /var/opt/opscode/local-mode-cache/chef-stacktrace.out
[2018-09-12T12:51:13-04:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2018-09-12T12:51:13-04:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2018-09-12T12:51:13-04:00] FATAL: Chef::Exceptions::MultipleFailures: Multiple failures occurred:
* Errno::ECONNREFUSED occurred in chef run: ruby_block[bootstrap-chef-server-data] (private-chef::bootstrap line 34) had an error: Errno::ECONNREFUSED: Failed to open TCP connection to 127.0.0.1:9463 (Connection refused - connect(2) for "127.0.0.1" port 9463)
* Mixlib::ShellOut::ShellCommandFailed occurred in delayed notification: runit_service[nginx] (private-chef::nginx line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx ----
STDOUT: fail: /opt/opscode/service/nginx: unable to change to service directory: file does not exist
STDERR: 
---- End output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx ----
Ran /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx returned 1

[2018-09-12T12:51:13-04:00] FATAL: Chef::Exceptions::MultipleFailures: Multiple failures occurred:
* Errno::ECONNREFUSED occurred in chef run: ruby_block[bootstrap-chef-server-data] (private-chef::bootstrap line 34) had an error: Errno::ECONNREFUSED: Failed to open TCP connection to 127.0.0.1:9463 (Connection refused - connect(2) for "127.0.0.1" port 9463)
* Mixlib::ShellOut::ShellCommandFailed occurred in delayed notification: runit_service[nginx] (private-chef::nginx line 37) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '1'
---- Begin output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx ----
STDOUT: fail: /opt/opscode/service/nginx: unable to change to service directory: file does not exist
STDERR: 
---- End output of /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx ----
Ran /opt/opscode/embedded/bin/sv restart /opt/opscode/service/nginx returned 1
```

## Steps to Reproduce (for bugs)

1. Install chef-server 12.17.33 on CentOS 6.10 or 7.5
1. Run `chef-server-ctl reconfigure`

## Your Environment

* Chef Server Version: 12.17.33
* Total/free RAM and disk space: 24GB/8GB RAM 412G/19G disk
* Operating System and Version: CentOS 6.10 & CentOS 7.5
* If upgrading, previous Chef Server version: N/A
* Running in a container?  LXC container.

**Comment by kevinreedy** [comment]
Having the same issue on Ubuntu 18.04

**Comment by markan** [comment]
May be related to #942

**Comment by markan** [comment]
TLDR; I can't repro  any problems on boxes with >=2GB of RAM. With 1GB of RAM, things fail terribly; SOLR wants 700MB+ of RAM or more. If that's it, then I think we can close this, as we generally recommend 4GB or more (https://docs.chef.io/chef_system_requirements.html).

However, if people are having troubles with higher memory instances we have a real regression, and I'd love to get details to repro. 

Long version:

I was able to successfully install and reconfigure chef-server on m5.large, t3.medium, and t3.small instances:

```
aws ec2 run-instances --region us-east-1  --image-id ami-0ac019f4fcb7cb7e6 --count 1 --instance-type t3.medium [... stuff deleted...]

# ssh to instance:
% wget 'https://packages.chef.io/files/stable/chef-server/12.18.14/ubuntu/18.04/chef-server-core_12.18.14-1_amd64.deb'
% sudo dpkg -i chef-server-core_12.18.14-1_amd64.deb 
% sudo chef-server-ctl reconfigure
```

**Comment by markan** [comment]
Re the original report; we can repro this issue on Ryan's LXC, but can't in AWS/centos 7.5/ ami-9887c6e7.

At this point we think it's LXC quirks.

**Comment by markan** [comment]
I'm going to close this issue for now.
